### PR TITLE
polygon-js-sdk new packer / js-iden3-auth

### PR DESCRIPTION
**Pull request by Kolezhniuk**

### CODE REVIEW

File: src/iden3comm/constants.ts [hunk 1]
```diff
@@ -38,5 +37,73 @@ export const PROTOCOL_MESSAGE_TYPE = Object.freeze({
...
+    /**
+     * @deprecated, supported for backward compatibility. Equivalent to EcdsaSecp256k1VerificationKey2019 when key is
+     *   not an ethereumAddress
+     */
+    'Secp256k1SignatureVerificationKey2018',
+    /**
+     * @deprecated, supported for backward compatibility. Equivalent to EcdsaSecp256k1VerificationKey2019 when key is
+     *   not an ethereumAddress
+     */
+    'EcdsaPublicKeySecp256k1',
```
Comment (vmidyllic) [inline src/iden3comm/constants.ts:71]: Remove EcdsaPublicKeySecp256k1

File: src/iden3comm/constants.ts [hunk 2]
```diff
@@ -38,5 +37,73 @@ export const PROTOCOL_MESSAGE_TYPE = Object.freeze({
...
+
+/**
+ * Universal resolver url
+ */
+export const UNIVERSAL_RESOLVER_URL = 'https://dev.uniresolver.io/1.0/identifiers';
+
+export const SUPPORTED_PUBLIC_KEY_TYPES = {
+  ES256: ['JsonWebKey2020'],
+  ES256K: [
+    'EcdsaSecp256k1VerificationKey2019',
```
Comment (vmidyllic) [inline src/iden3comm/constants.ts:52]: we need to handle ethereumAddress (if ethereumAddress - > also use custome singer)

File: src/iden3comm/constants.ts [hunk 3]
```diff
@@ -38,5 +37,73 @@ export const PROTOCOL_MESSAGE_TYPE = Object.freeze({
...
+    'Secp256k1SignatureVerificationKey2018',
+    /**
+     * @deprecated, supported for backward compatibility. Equivalent to EcdsaSecp256k1VerificationKey2019 when key is
+     *   not an ethereumAddress
+     */
+    'EcdsaPublicKeySecp256k1',
+    'JsonWebKey2020'
+  ],
+  'ES256K-R': [
+    'EcdsaSecp256k1VerificationKey2019',
```
Comment (vmidyllic) [inline src/iden3comm/constants.ts:75]: remove deprecated

File: src/iden3comm/constants.ts [hunk 4]
```diff
@@ -38,5 +37,73 @@ export const PROTOCOL_MESSAGE_TYPE = Object.freeze({
...
+     */
+    'Secp256k1SignatureVerificationKey2018',
+    /**
+     * @deprecated, supported for backward compatibility. Equivalent to EcdsaSecp256k1VerificationKey2019 when key is
+     *   not an ethereumAddress
+     */
+    'EcdsaPublicKeySecp256k1',
+    'JsonWebKey2020'
+  ],
+  Ed25519: [
```
Comment (vmidyllic) [inline src/iden3comm/constants.ts:97]: remove Ed25519

File: src/iden3comm/constants.ts [hunk 5]
```diff
@@ -38,5 +37,73 @@ export const PROTOCOL_MESSAGE_TYPE = Object.freeze({
...
+    'ED25519SignatureVerification',
+    'Ed25519VerificationKey2018',
+    'Ed25519VerificationKey2020',
+    'JsonWebKey2020'
+  ],
+  EdDSA: [
```
Comment (vmidyllic) [inline src/iden3comm/constants.ts:103]: remove EdDSA

File: src/iden3comm/packers/jws.ts [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+/**
+ * Packer that can pack message to JWZ token,
+ * and unpack and validate JWZ envelope
+ * @exports
+ * @beta
+ * @class ZKPPacker
+ * @implements implements IPacker interface
+ */
+export class JWSPacker implements IPacker {
+  // readonly vmPubkeyExtractorHandlerMap = {
```
Comment (vmidyllic) [inline src/iden3comm/packers/jws.ts:26]: remove comments

File: src/iden3comm/packers/jws.ts [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+      throw new Error('DID in params does not match DID in message');
+    }
+
+    if (!from) {
+      throw new Error('Missing DID');
+    }
+
+    const vmTypes: string[] = SUPPORTED_PUBLIC_KEY_TYPES[params.alg];
+    if (!vmTypes?.length) {
+      throw new Error(`No supported signature types for algorithm ${params.alg}`);
```
Comment (vmidyllic) [inline src/iden3comm/packers/jws.ts:72]: No supported verification methods for ..

File: src/iden3comm/packers/jws.ts [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+
+    // || vm.blockchainAccountId === params.kid
+
+    if (!vm) {
+      throw new Error(
+        `No key found with id ${params.kid} in ${section} section of DID document ${didDocument.id}`
+      );
+    }
+
+    if (vm.blockchainAccountId && !params.signer) {
```
Comment (vmidyllic) [inline src/iden3comm/packers/jws.ts:103]: if !vm.publicKeyHex &&  vm.ethereumAddress || vm.blockchainAccounId

Comment (vmidyllic) [reply src/iden3comm/packers/jws.ts:103]: maybe firstly we can do       const { publicKeyBytes, kmsKeyType } = extractPublicKeyBytes(vm); 
and if we don't have a hex -check this condition

File: src/kms/store/types.ts [hunk 9]
```diff
@@ -6,7 +6,10 @@
  */
 export enum KmsKeyType {
   BabyJubJub = 'BJJ',
-  Ethereum = 'ETH'
+  Ethereum = 'ETH',
+  Secp256k1 = 'Secp256k1',
+  Secp256r1 = 'Secp256r1',
+  Ed25519 = 'Ed25519'
```
Comment (vmidyllic) [inline src/kms/store/types.ts:12]: is it needed? we don't have a provider for it.

**Review by vmidyllic — CHANGES_REQUESTED** [review]

File: src/iden3comm/constants.ts (hunk 2 above)
Comment (Kolezhniuk) [reply src/iden3comm/constants.ts:52]: This map is about supported public key types, handling of ethereumAddress in jws packer

Comment (Kolezhniuk) [reply src/iden3comm/constants.ts:52]: This map is about supported public key types, handling of ethereumAddress in jws packer

**Comment by Kolezhniuk** [comment]
@vmidyllic fixed comments

**Review by vmidyllic — APPROVED** [review]

**Review by vmidyllic — APPROVED** [review]

**Pull request merged**